core/task/class.go

```diff
@@ -81,47 +82,104 @@ func (c *Class) UnmarshalYAML(unmarshal func(interface{}) error) (err error) {
 			}
 		}
 		*c = Class{
-			Identifier: aux.Identifier,
-			Defaults:   gera.MakeStringMapWithMap(aux.Defaults),
-			Control:    aux.Control,
-			Command:    aux.Command,
-			Wants:      aux.Wants,
-			Bind:       aux.Bind,
-			Properties: gera.MakeStringMapWithMap(aux.Properties),
-			Constraints:aux.Constraints,
-			Connect:    aux.Connect,
+			Identifier:  aux.Identifier,
+			Defaults:    gera.MakeStringMapWithMap(aux.Defaults),
+			Control:     aux.Control,
+			Command:     aux.Command,
+			Wants:       aux.Wants,
+			Bind:        aux.Bind,
+			Properties:  gera.MakeStringMapWithMap(aux.Properties),
+			Constraints: aux.Constraints,
+			Connect:     aux.Connect,
 		}
 	}
 	return
 
 }
 
-type taskClassIdentifier struct {
+func (c *Class) MarshalYAML() (interface{}, error) {
+	type _bind struct {
+		Name        string                `yaml:"name"`
+		Type        channel.ChannelType   `yaml:"type"`
+		SndBufSize  int                   `yaml:"sndBufSize"`
+		RcvBufSize  int                   `yaml:"rcvBufSize"`
+		RateLogging string                `yaml:"rateLogging"`
+		Transport   channel.TransportType `yaml:"transport"`
+		Addressing  channel.AddressFormat `yaml:"addressing"`
+	}
+
+	type _class struct {
+		Name     string            `yaml:"name"`
+		Defaults map[string]string `yaml:"defaults"`
+		Control  struct {
+			Mode string `yaml:"mode"`
+		} `yaml:"control"`
+		Wants       ResourceWants           `yaml:"wants"`
+		Bind        []_bind                 `yaml:"bind"`
+		Properties  map[string]string       `yaml:"properties"`
+		Constraints []constraint.Constraint `yaml:"constraints,omitempty"`
+		Command     *common.CommandInfo     `yaml:"command"`
+	}
+	aux := _class{
+		Name:        c.Identifier.Name,
+		Defaults:    c.Defaults.Raw(),
+		Properties:  c.Properties.Raw(),
+		Wants:       c.Wants,
+		Constraints: c.Constraints,
+		Command:     c.Command,
+	}
+
+	// Flatten bind struct to have channel elements and addressing together
+	for _, bind := range c.Bind {
+		auxBind := _bind{
+			Name:        bind.Name,
+			Type:        bind.Type,
+			SndBufSize:  bind.SndBufSize,
+			RcvBufSize:  bind.RcvBufSize,
+			RateLogging: strconv.Itoa(bind.RateLogging),
```

Comment:
These 3 are optional, if the DPL dump doesn't provide explicit values, we can simply omit them from the YAML output. This is not mandatory to fix now, you can make a JIRA ticket so we don't forget about it.

Reply:
Considering these are currently not present in a DPL dump, how does defaulting them to empty values and adding a `yaml:"omitempty"` to the Marshaler sound? We can simply replace the default values as and when we get an output from the dump.

Reply:
Agreed. This kind of information should be filled either by an explicit setting in the DPL dump, or by default values passed by the user.